Add namespace to forms

fkm opened the pull request:
Plus, fixes a problem with the validation.

Resolves #114

czosel commented on addon/components/cfb-form-editor/general.js:
```diff
@@ -20,11 +21,40 @@ export default Component.extend(ComponentQueryManager, {
+      parent.classList.add("slugnamespace");
+      parent.insertBefore(span, input);
+    }
+  },
```
Is this _exactly_ the same as for questions? If yes, it might be worth extracting this into a shared function?

fkm replied:
It is. But I'd rather concentrate my efforts on writing the custom component.

czosel replied:
Alright!

fkm commented:
Coverage is lacking. Somehow the component is still in its loading state when the `addSlug` method is invoked. This should not be the case as I changed the `didReceiveAttrs` lifecycle hook to `async` and added an `await` to the `.perform()`.

https://github.com/projectcaluma/ember-caluma/pull/115/files#diff-68e6340d2cf7a6ed2819d387bc84e37dR50

Works outside of testing and works in testing in `question.js` where I did the same thing. As I intend to write a custom component for the prefixed slug, I recommend to ignore the missing coverage in this case.

fkm commented:
As this only fails in `ember-canary` and only on some code that will soon be removed by another PR I recommend that we merge this one despite the failed test.

czosel commented:
Hmm, looks like the test with floating deps is failing, or am I seeing something wrong?

fkm commented:
Superseded by #116
